ee/desktop/menu/menu.go

```diff
@@ -1,36 +1,174 @@
 package menu
 
 import (
-	"fmt"
+	"os/exec"
+	"runtime"
 
 	"fyne.io/systray"
-	"github.com/kolide/kit/version"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
 	"github.com/kolide/launcher/ee/desktop/assets"
 )
 
-func Init(hostname string) {
-	onReady := func() {
-		systray.SetTemplateIcon(assets.KolideDesktopIcon, assets.KolideDesktopIcon)
-		systray.SetTooltip("Kolide")
+// MenuData encapsulates a systray icon and accessible menu items
+type MenuData struct {
+	Tooltip string         `json:"tooltip,omitempty"`
+	Items   []MenuItemData `json:"items"`
+}
 
-		versionItem := systray.AddMenuItem(fmt.Sprintf("Version %s", version.Version().Version), "")
-		versionItem.Disable()
+// MenuItemData represents a menu item, optionally containing sub menu items
+type MenuItemData struct {
+	Label       string         `json:"label,omitempty"`
+	Tooltip     string         `json:"tooltip,omitempty"`
+	Disabled    bool           `json:"disabled,omitempty"`
+	NonProdOnly bool           `json:"nonProdOnly,omitempty"`
+	IsSeparator bool           `json:"isSeparator,omitempty"`
+	Action      MenuItemAction `json:"action,omitempty"`
+	Items       []MenuItemData `json:"items,omitempty"`
+}
 
-		// if prod environment, return
-		if hostname == "k2device-preprod.kolide.com" || hostname == "k2device.kolide.com" {
-			return
-		}
+// MenuItemAction encapsulates what action should be performed when a menu item is invoked
+type MenuItemAction struct {
+	URL string `json:"url,omitempty"`
+}
+
+// MenuBuilder is an interface a menu parser can use to specify how the systray menu is built
+type MenuBuilder interface {
+	SetTooltip(tooltip string)
+	AddMenuItem(label, tooltip string, disabled, nonProdOnly bool, action MenuItemAction, parent any) any
+	AddSeparator()
+}
+
+type menu struct {
+	logger    log.Logger
+	hostname  string
+	doneChans []chan bool
```

Comment:
absolutely not important, but you can use a `chan struct{}` if you just signalling and don't want to send any data over channel it consumes no memory